components/SearchBar.vue

```diff
@@ -0,0 +1,107 @@
+<template>
+  <form class="mt-8 md:mt-0 md:flex md:items-center md:justify-between md:flex-wrap md:p-6">
+    <div class="md:flex md:items-center md:flex-shrink-0 md:text-white md:mr-6">
+      <Logo class="logo inline"/>
+      <span class="ml-4 text-2xl text-accent-purple">AutoLizing</span>
+      <button class="hamburger md:hidden" type="button" @click="menuOpen = !menuOpen">
+        <div class="hamburger-line  bg-grey-600"></div>
+        <div class="hamburger-line  bg-grey-600"></div>
+        <div class="hamburger-line  bg-grey-600"></div>
+      </button>
+      <div class="dropdown mt-6 md:hidden" :class="{ 'dropdown-after' : menuOpen }">
+        <input type="text" placeholder="Vehicle search" class="dropdown-search w-8/12 bg-grey-200 p-2 rounded-md inline-block text-black">
+        <button class="bg-accent-purple inline-block btn-search"><Search class="btn-search_icon"/></button>
+      </div>
+    </div>
+    <div class="w-full block md:flex-grow md:flex md:items-center md:w-auto">
+      <div class="text-xs hidden md:block md:flex-grow">
+        <input type="text" placeholder="Vehicle search" class="big-input bg-grey-200 px-4 h-md-40 rounded-l">
+        <button class="bg-accent-purple  btn-search"><Search class="btn-search_icon"/></button>
+      </div>
+      <div class="hidden md:block">
+        <button class="btn btn-purple">To add an advert</button>
+      </div>
+    </div>
+  </form>
+</template>
+
+<script>
+  import Search from '~/assets/svg/search.svg?inline';
+  import Logo from '~/assets/svg/logo.svg?inline';
+    export default {
+      components: {Logo, Search},
+      name: "SearchBar",
+      data() {
+        return {
+          menuOpen: false
+        }
+      }
+    }
+</script>
+
+<style lang="scss" scoped>
+  .hamburger {
+    position: absolute;
+    margin-left: 3rem;
+    margin-top: 9px;
+    cursor: pointer;
+  &:focus {
+     outline: none;
+   }
+  &-line {
+     margin-bottom: 8px;
+     width: 30px;
+     height: 2px;
+   }
+  &-line:last-child {
+     margin-bottom: 0;
+   }
+  }
+  .dropdown {
+    height: 0px;
+    width: 104%;
```

Comment:
It should take all free space (width), when wrapper container will has margin right the same as from left.